models/Import/Service/QtiTestItemsService.php

```diff
@@ -0,0 +1,105 @@
+use oat\taoQtiTest\models\runner\QtiRunnerService;
+use oat\taoQtiTest\models\runner\QtiRunnerServiceContext;
+use qtism\data\AssessmentTest;
+use qtism\data\ExtendedAssessmentItemRef;
+use qtism\data\ExtendedAssessmentSection;
+use qtism\data\TestPart;
+use taoQtiTest_helpers_Utils;
+
+class QtiTestItemsService
+{
+    private QtiRunnerService $qtiRunnerService;
+    private DeliveryExecutionManagerService $deliveryExecutionService;
+    private DeliveryContainerService $deliveryContainerService;
+
+    public function __construct(
+        QtiRunnerService $qtiRunnerService,
+        DeliveryExecutionManagerService $deliveryExecutionService,
+        DeliveryContainerService $deliveryContainerService,
+    ) {
+
+        $this->qtiRunnerService = $qtiRunnerService;
+        $this->deliveryExecutionService = $deliveryExecutionService;
+        $this->deliveryContainerService = $deliveryContainerService;
+    }
+
+    public function getItemsByDeliveryExecutionId(string $deliveryExecutionId): array
+    {
+        $context = $this->getServiceContext($deliveryExecutionId);
+        $testDefinition = $this->getDefinition($context);
+        $testParts = [];
+        /** @var TestPart $testPart */
+        foreach ($testDefinition->getTestParts() as $testPart) {
+            $sections = [];
+            /** @var ExtendedAssessmentSection $section */
+            foreach ($testPart->getAssessmentSections() as $section) {
+                $items = [];
+                /** @var ExtendedAssessmentItemRef $item */
+                foreach ($section->getSectionParts() as $item) {
+                    $itemData = $this->qtiRunnerService->getItemData($context, $item->getHref());
+                    $itemData['data']['isExternallyScored'] = $this->isExternallyScored($itemData['data'] ?? []);
+
+                    $items[$item->getIdentifier()] = $itemData['data'];
+                }
+                $sectionId = $section->getIdentifier();
+                $sections[$sectionId] = $items;
+            }
+            $testPartId = $testPart->getIdentifier();
+            $testParts[$testPartId] = $sections;
+        }
+
+        return $testParts;
+    }
+
+    protected function getDefinition(QtiRunnerServiceContext $context): AssessmentTest
+    {
+        return taoQtiTest_helpers_Utils::getTestDefinition($context->getTestCompilationUri());
+    }
+
+    protected function getServiceContext($deliveryExecutionId): QtiRunnerServiceContext
+    {
+        $deliveryExecution = $this->deliveryExecutionService->getDeliveryExecutionById($deliveryExecutionId);
+
+        $compilation = $this->deliveryContainerService->getTestCompilation($deliveryExecution);
+        $testId = $this->deliveryContainerService->getTestDefinition($deliveryExecution);
+
+        return $this->qtiRunnerService->getServiceContext($testId, $compilation, $deliveryExecutionId);
+    }
+
+    private function isExternallyScored(array $data): bool
```

Comment:
not needed anymore

Reply:
Done

Reply:
I do not see

Reply:
I removed isExternallyScored() as we discussed